src/fhir/api.test.ts

```diff
@@ -7,8 +7,92 @@ import { geneCoding } from "../code_systems/hgnc";
 const fhir = new Fhir();
 
 const reportedGenes = [geneCoding("HGNC:4389", "GNA01")];
+const FHIR_URL = process.env.REACT_APP_FHIR_URL || "";
+
+const getPatients = async () => {
+  const url = `${FHIR_URL}/Patient`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    console.error(response.statusText);
+    console.error(response.body);
+    throw new Error(response.statusText);
+  }
+  return await response.json();
+};
+
+const check = async (response: any) => {
+  // would like to figure out which type response should be
+  const r = await response.json();
+  console.error(r.body);
+};
+
+const deletePatients = async (patientId?: string) => {
+  /**
+   * This method can take a string to delete a particular patient's records
+   * or it can accept a bundle with a number of entries from which
+   * it will extract the ID(s) and delete the patient(s)
+   */
+  const baseURL = `${FHIR_URL}/Patient`;
+  const patientData = await getPatients();
+  if (!("entry" in patientData)) {
+    return;
+  }
+
+  const deletePatientEntry = async (entry: any) => {
+    let id;
+    if (typeof entry === "string") {
+      console.info("entry is string", patientId);
+      return (id = patientId);
+    }
+    id = entry[0].resource.id;
+    console.info("deleting records related to " + id);
+    const response = await fetch(`${baseURL}/${id}?_cascade=delete`, {
+      method: "DELETE",
+    });
+    if (!response.ok) {
+      console.error(response.statusText);
+      console.error(response.body);
+    }
+    check(response);
+    await new Promise((r) => setTimeout(r, 1500));
+  };
+
+  const patientEntry = patientData.entry;
+  deletePatientEntry(patientEntry);
+};
 
 describe("FHIR resources", () => {
+  beforeEach(async () => {
+    fetchMock.dontMock();
+    await deletePatients();
+  });
+
+  test("database is clear on setup", async () => {
+    /**
+     * Before doing tests on the database, we want to clear all its data
+     */
```

Comment:
❤️ for documenting the test, should be above the test method to follow javascript documentation conventions
```suggestion
    /**
     * Before doing tests on the database, we want to clear all its data
     */
  test("database is clear on setup", async () => {
```

Reply:
whoops, didn't know about this 🤭